recipes/date/all/conanfile.py

```diff
@@ -1,29 +1,91 @@
 import os
-from conans import ConanFile, tools
+from conans import ConanFile, CMake, tools
 
 
 class DateConan(ConanFile):
     name = "date"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/HowardHinnant/date"
     description = "A date and time library based on the C++11/14/17 <chrono> header"
-    topics = ("date", "datetime", "timezone", "calendar", "time", "iana-database")
+    topics = ("date", "datetime", "timezone",
+              "calendar", "time", "iana-database")
     license = "MIT"
-    exports_sources = ["patches/*"]
-    no_copy_source = True
+    exports_sources = ["patches/*", "CMakeLists.txt"]
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "use_system_tz_db": [True, False],
+               "use_tz_db_in_dot": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "use_system_tz_db": True,
+                       "use_tz_db_in_dot": False}
+
+    _cmake = None
 
     @property
     def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
+        return "source_subfolder"
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        #requires at least c++11 to work.
+        tools.check_min_cppstd(self, "11")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        cmake = CMake(self)
+        cmake.definitions["ENABLE_DATE_TESTING"] = False
+        cmake.definitions["USE_SYSTEM_TZ_DB"] = self.options.use_system_tz_db
+        cmake.definitions["USE_TZ_DB_IN_DOT"] = self.options.use_tz_db_in_dot
+        cmake.configure()
+
+        self._cmake = cmake
+        return self._cmake
+
+    def requirements(self):
+        if not self.options.use_system_tz_db:
+            self.requires("libcurl/7.67.0")
```

Comment:
I doesn't work with latest version of libcurl available in CCI? `libcurl/7.69.1`

Reply:
There are no tests testing that functionality at all. 
Is there any possibility in the `conan test` in this CI to actually test a matrix of option rather than fixed configuration?

Reply:
Applied, I have *not* checked whether it works on windows.

Reply:
it works, I've checked